### PR TITLE
feat(nohardlinks): add category directory ignore

### PR DESCRIPTION
**Pull request by jacobcxdev**
## Summary

- Replaces #865 on current `develop` with a per-category `ignore_category_dir` option.
- Defaults the option to `false`, preserving existing configurations.
- Gives `ignore_root_dir` precedence. Otherwise, category-local links are ignored only when `ignore_category_dir` is enabled; links outside that category still count.
- Updates configuration parsing, documentation, sample configuration, Web UI schema and save handling, and test fakes.

## Why this differs from `ignore_root_dir`

`ignore_root_dir` subtracts every link under the configured root. That cannot distinguish a cross-seed category from another download category under the same root. Category scope allows cross-seeds to ignore one another while still detecting a source or library link outside their category.

## Upstream reconciliation

- Extends the current compact-logging and cached-stat implementation without a second filesystem traversal.
- Counts filesystem identities as `(st_dev, st_ino)` to prevent cross-device inode collisions.
- Preserves folder size ordering, threshold and early exits, return values, counters, Windows/UNC boundaries, remote path mapping, and first-declared ties.
- Supports multiple paths, nested/overlapping paths, glob scopes, and glob-matched ancestor directories using the most-specific matched scope.
- PRs #1159 and #1160 remain open and unmerged, so this does not depend on their proposed global-options or unknown-key-warning machinery. The new setting remains a normal per-category field compatible with that direction.

## Previous review and closure

#865 was initially closed because `ignore_root_dir` was thought to cover the use case, then reopened after the category-versus-root distinction was demonstrated. Its remaining review asked for a cleaner sample entry, consistent option lookup, retained unrelated type annotations, and stronger runtime evidence. It was ultimately closed for inactivity on 27 February 2026.

This replacement starts from current `develop`, leaves unrelated annotations untouched, cleans up the sample, uses one consistent backward-compatible default, and supplies automated coverage for the requested behaviour and edge cases.

## Verification

- [x] `uv run pytest -q` — 397 passed
- [x] `uv run ruff check .`
- [x] `uv run ruff format --check .`
- [x] `uv run pre-commit run --all-files`
- [x] `npm run test:web-ui` — 19 passed
- [x] Independent code review completed with no remaining Critical or Important findings

### CODE REVIEW

File: config/config.yml.sample [hunk 1]
```diff
@@ -162,7 +162,10 @@ nohardlinks:
       - Beyond-HD
       - BroadcasTheNet
     # <OPTIONAL> ignore_root_dir var: Will ignore any hardlinks detected in the same root_dir (Default True).
-    ignore_root_dir: true
+    ignore_root_dir: false
```
Comment (bakerboy448) [inline config/config.yml.sample]: Why are we making a breaking default change?

Comment (jacobcxdev) [reply config/config.yml.sample]: Good catch. Restored `ignore_root_dir: true` and set the new `ignore_category_dir` sample option to its backward-compatible `false` default in 33da0a4. This preserves existing sample behaviour while documenting the new option.
